public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -486,12 +553,48 @@ export class PrometheusDatasource extends DataSourceApi<PromQuery, PromOptions>
     return { __rate_interval: { text: rateInterval + 's', value: rateInterval + 's' } };
   }
 
+  processIntervalV2(target: PromQuery, options: DataQueryRequest<PromQuery>) {
```

Comment:
This is the same logic as is in [createQuery](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/prometheus/datasource.ts#L407-L451)